src/components/simulator/input/d3Chart/utils.ts

```diff
@@ -1,9 +1,20 @@
 import { max, min, select } from 'd3';
 import { ChartPrices } from 'components/simulator/input/d3Chart/D3ChartCandlesticks';
 import { CandlestickData } from 'libs/d3/types';
+import { useEffect, useState } from 'react';
 
 export const getSelector = (selector: string) => select(`.${selector}`);
 
+export const useSelectable = (selector: string) => {
+  const [isSelectable, setIsSelectable] = useState(false);
+  useEffect(() => {
+    const selection = getSelector(selector);
+    if (!selection.size()) return;
```

Comment:
question: shouldn't there be a code path that sets it to false? e.g. when the selection size is 0?

Reply:
The selection size is 0 when D3 doesn't find the SVGElement. It's a way to wait for the element to be rendered in the DOM.